core/vm/evm.go

```diff
@@ -137,18 +151,23 @@ type EVM struct {
 	// available gas is calculated in gasCall* according to the 63/64 rule and later
 	// applied in opCall*.
 	callGasTemp uint64
+
+	// AtomicTransactor is used to create atomic transactions in shared memory for the purpose of
+	// cross-chain transfers.
+	AtomicTransactor AtomicTransactor
 }
 
 // NewEVM returns a new EVM. The returned EVM is not thread safe and should
 // only ever be used *once*.
-func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig *params.ChainConfig, config Config) *EVM {
+func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig *params.ChainConfig, config Config, atomicTransactor AtomicTransactor) *EVM {
```

Comment:
`atomicTransactor AtomicTransactor` appears in various constructors and objects to thread its way through to the EVM and state processor